packages/core/src/mesh/ModelMesh.ts

```diff
@@ -14,4 +67,693 @@ export class ModelMesh extends Mesh {
     super(engine);
     this.name = name;
   }
+
+  /**
+   * Upload Mesh Data to the graphics API.
+   * @param noLongerAccessible - Whether to access data later. If true, you'll never access data anymore (free memory cache)
+   */
+  uploadData(noLongerAccessible: boolean): void {
+    const { _indices, _vertexSlotChanged, _accessible } = this;
+    if (!_accessible) {
+      throw "Not allowed to access data while accessible is false.";
+    }
+
+    // Structure of vertex buffer has changed
+    if (_vertexSlotChanged || this._vertexCountChanged) {
+      const vertexElements = this._updateVertexElements();
+      const elementCount = this._elementCount;
+      let vertices = this._verticesFloat32;
+      if (!vertices || vertices.length !== elementCount * this._vertexCount) {
+        vertices && this._vertexBuffer.destroy();
+        vertices = new Float32Array(elementCount * this._vertexCount);
+        this._verticesUint8 = new Uint8Array(vertices.buffer);
+        this._verticesFloat32 = vertices;
+      }
+      this._vertexChangeFlag = ValueChanged.All;
+      this._updateVertices(vertices);
+      this._vertexBuffer = new Buffer(
```

Comment:
```this._vertexBuffer = new Buffer()``` should inside ```if (!vertices || vertices.length !== elementCount * this._vertexCount) ```,only Buffer byteLength change need re-construct

Reply:
+1